src/software/new_geom/convex_shape.h

```diff
@@ -0,0 +1,8 @@
+#include "software/new_geom/shape.h"
+
+/**
+ * A non-polygon shape that is convex (curved outwards).
```

Comment:
Is this a non-polygon shape? What if we have `Square : public ConvexShape, public Polygon`? I think a square is both a valid polygon and a convex shape?

Reply:
Fixed in latest, forgot this will be inherited by ```Polygon```